app_dart/lib/src/foundation/utils.dart

```diff
@@ -182,11 +182,22 @@ List<String> validateOwnership(String ciYamlContent, String testOwnersContent, {
 
   final pb.SchedulerConfig schedulerConfig = ciYamlFromProto.config;
 
+  // TODO this is where to enforce task name for devicelab.
   for (pb.Target target in schedulerConfig.targets) {
-    final String builder = target.name;
-    final String? owner = getTestOwnership(
+    String builder = target.name;
+    final BuilderType builderType = getTypeForBuilder(
       builder,
-      getTypeForBuilder(builder, ciYamlFromProto, unfilteredTargets: unfilteredTargets),
+      ciYamlFromProto,
+      unfilteredTargets: unfilteredTargets,
+    );
+
+    if (builderType == BuilderType.devicelab) {
+      builder = target.properties['task_name']!;
+    }
```

Comment:
Do we need this block?

Reply:
Good catch. I don't think we do. I will test.